### PR TITLE
Accept input as an array, string, literal or object

### LINKED ISSUE #13: Document and improve the function interface
Since I use Oceanwind through a function (not a template), I've ended up with a pattern like this:

```js
ow([combinedClasses.join(" ")])
```

`combinedClasses` is an array of classes and to comply with the current API, I have to transform it. Maybe it would be better to have Oceawind to detect the case and work directly with an array like this:

```js
ow(combinedClasses)
```

It's a bit counter-intuitive to add the array wrapping. Maybe a check or so would be possible.

### PR DESCRIPTION
**Pull request by lukejacksonn**
Fixes #13 

Perviously it was only possible to invoke oceanwind via a tagged template literal with no dynamic parts (or mimicking this with `oceanwind([rules])`). The changes here allows for oceanwind to be invoked in 4 ways:

1. Tag Template Literal

```js
ow`bg-red-500 rounded`;
ow`bg-red-500 ${false && 'rounded'}`; // falsey interpolations will be omitted
```

2. Function call passing a string

```js
ow('bg-red-500 rounded');
```

3. Function call passing an array

```js
ow(['bg-red-500', 'rounded']);
ow(['bg-red-500', false && 'rounded']); // falsey items will be omitted
```

4. Function call passing an object

```js
ow({ 'bg-red-500': true, rounded: false }); // keys with falsey values will be omitted
```

The first improvement is the dynamic parts. The second, being able to pass strings without having to wrap in an array. The last is being able to pass rules or rule sets that are to be applied conditionally (kinda like [`clsx`](https://github.com/lukeed05/clsx)).

Still need to update the README to document this change.

### CODE REVIEW

**Comment by bebraw** [comment]
Does the second option cover for applying multiple class names? I have an array of strings and I then apply Oceanwind against that. I imagine it would be `ow(classes.split(' '))` which isn't bad.

If that's possible, big 👍 then.

**Comment by lukejacksonn** [comment]
I was trying to account for your case whereby you were inputting `classes.join(' ')` but perhaps I misunderstood and actually you wanted to input an array without having to join it into a string.

I'm already using an array check to infer a template literal being used, so it might conflict 🤔

**Comment by bebraw** [comment]
@lukejacksonn Yeah, so at the moment I do some tricks to adapt to your API. The ideal would be if it accepted an array of strings out of the box but I can also resolve my data to any form that the API accepts.

From my point of view, `ow(['bg-red-500', 'p-4'])` would likely be the ideal as then I wouldn't have to transform the data at all.

**Comment by lukejacksonn** [comment]
Ok, well it turns out I got this all wrong anyway so I'm going to start a fresh. Will be back with a solution shortly.

**Comment by bebraw** [comment]
@lukejacksonn No worries, ping me if you need any help. 👍

File: README.md [hunk 1]
```diff
@@ -35,14 +35,14 @@ import ow from 'https://unpkg.com/oceanwind';
...
-The code above will result in the following happening:
+Running the above code will result in the following happening:
 
 1. Shorthand syntax will be translated into CSS (e.g. `h-full => { height: 100vh }`).
 2. All resultant CSS will be merged into a single CSS-in-JS object
 3. Each style will be assigned a unique class and appended to a stylesheet
 4. A string is returned representing all the classes that were created
 
-In addition to importing oceanwind to generate class names for given shorthand. It is reccomended to import two static css files which help normalize styles across browsers. These are:
+It is reccomended to import the following css files which help normalize styles across browsers:
```
Comment (bebraw) [inline README.md:46]: /s/reccomemded/recommended

**Comment by bebraw** [comment]
The proposed API looks great.

**Comment by lukejacksonn** [comment]
Great! Well I'm happy with the additional flexibility and it wasn't that hard to support all these other variants either. Thanks for creating the issue and reviewing 🙇

**Pull request merged**